src/transport_ofi.c

```diff
@@ -1383,7 +1348,15 @@ int shmem_transport_init(void)
         }
         shmem_transport_ofi_stx_max = 1;
     } else {
-        shmem_transport_ofi_stx_max = shmem_internal_params.OFI_STX_MAX;
+        if (shmem_internal_params.OFI_STX_MAX_provided &&
+            shmem_internal_params.OFI_STX_MAX <= 0) {
+            RAISE_WARN_MSG("Ignoring non-positive OFI_STX_MAX value '%ld', enabling OFI_STX_AUTO\n",
+                           shmem_internal_params.OFI_STX_MAX);
+            /* FIXME: Setting auto here could be bad ... better to set 1 or error? */
+            shmem_internal_params.OFI_STX_AUTO = 1;
```

Comment:
Need to revisit this FIXME

Reply:
I think I prefer to raise an error in this case.  @davidozog what's your preference?

Reply:
Yeah, an error here makes sense.